
# Launch `pytest` from Terminal or with `pytest.main` different behaviour (importing modules)

Here is a minimal reproducible example. I have three files, organised like those:

In unit_tests.py, I have:
import pytest

retcode: int | pytest.ExitCode = pytest.main()

if retcode == pytest.ExitCode.OK:
    print("Good")

In my_aux_file.py, I have:
def main():
    print("Hello")

In test_file.py, I have:
from my_aux_file import main

def test_main() -> None:
    main()

Why does, running pytest in the terminal (in the parent folder pytest_bug) returns ModuleNotFoundError: No module named 'my_aux_file' and running python unit_tests.py returns :
platform win32 -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\lhott\Desktop\pytest_bug
plugins: mypy-0.10.0
collected 1 item

my_folder\test_file.py .                                                                                         [100%]

================================================== 1 passed in 0.01s ==================================================

Does pytest.main() do something that fixes the import bug I had when just running pytest in the terminal? Does pytest change the source folder even though I launched it in pytest_bug? I would expect the source folder / root directory to be set where the command pytest is written, right?

        